js/adapt-devtools.js

```diff
@@ -278,6 +280,48 @@ define([
 			Adapt.trigger('drawer:closeDrawer');
 		},
 
+		/*************************************************/
+		/***************** COMPLETE MENU *****************/
+		/*************************************************/
+
+		_checkCompleteMenuVisibility:function() {
+			var currentModel = Adapt.findById(Adapt.location._currentId);
+
+			if (currentModel.get('_type') != 'menu' && currentModel.get('_type') !== "course" ) {
+				this.$('.complete-menu').addClass('display-none');
+				return;
+			}
+
+			var cond = currentModel.has('_isInteractionsComplete') ? {'_isInteractionsComplete':false} : {'_isInteractionComplete':false};
+			var incomplete = currentModel.findDescendants('components').where(cond);
+
+			this.$('.complete-menu').toggleClass('display-none', incomplete.length == 0);
+
+		},
+
+		onCompleteMenu:function(e) {
+			var currentModel = Adapt.findById(Adapt.location._currentId);
+
+			if (Adapt.devtools.get('_trickleEnabled')) Adapt.trigger("trickle:kill");
+
+			var cond = currentModel.has('_isInteractionsComplete') ? {'_isInteractionsComplete':false} : {'_isInteractionComplete':false};
+			var incomplete = currentModel.findDescendants('components').where(cond);
+
+			_.each(incomplete, function(component) {
+				if (component.get('_isQuestionType')) {
+					component.set("_isCorrect", true);
+					component.set("_isSubmitted", true);
+					component.set("_score", 1);
+					component.set("_attemptsLeft", Math.max(0, component.set("_attempts") - 1));
+				}
+				
+				component.set("_isComplete", true);
+				component.set(currentModel.has('_isInteractionsComplete') ? '_isInteractionsComplete' : '_isInteractionComplete', true);
```

Comment:
This check can be dropped now that the framework dependency is >= 2.2: only isInteractionComplete is relevant